automation/MPLogger.py

```diff
@@ -265,10 +263,21 @@ def _handle_serialized_writes(self, obj):
         This is currently records that are written to a file on disk
         and those sent to Sentry.
         """
+        def ensure_str(s):
+            """Ensures we get a string
+
+            Raises a TypeError or UnicodeError if it's not
+            """
+            if not isinstance(s, (str, bytes)):
+                raise TypeError("not expecting type '%s'" % type(s))
+            elif isinstance(s, bytes):
+                s = s.decode('utf-8', 'strict')
+            return s
+
         if obj['exc_info']:
-            obj['exc_info'] = dill.loads(six.ensure_str(obj['exc_info']))
+            obj['exc_info'] = dill.loads(ensure_str(obj['exc_info']))
```

Comment:
@englehardt Has this ever worked?
Dill says it converts objects into an arbitrary byte stream.
So from my understanding we would need to do it the other way around.
```suggestion
            obj['exc_info'] = ensure_str(dill.loads(obj['exc_info']))
```

Reply:
This code was introduced [here](https://github.com/mozilla/OpenWPM/commit/8a4a293ab4f58cd5885ce3067d6629df374090c3)

Reply:
Just tried switching the order, that doesn't work either as exc_info contains a tuple, so I'm updating my suggestion to just remove `ensure_str` unless there's something I'm missing here.

```suggestion
            obj['exc_info'] = dill.loads(obj['exc_info'])
```

Reply:
after introducing some hacky logging I see this:
``` python
obj['exc_info'] is of type <class 'tuple'> :  (<class 'selenium.common.exceptions.WebDriverException'>, WebDriverException("invalid argument: can't kill an exited process", None, None), <traceback object at 0x7f1a36320188>)
```

Reply:
This would mean that we never sent exceptions to sentry, because we would always error out before sending them?

Reply:
I filed https://github.com/mozilla/OpenWPM/issues/534 to investigate why our tests failed to capture this.